packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -1013,23 +1032,26 @@ class SlideshowView extends React.Component {
     const renderGroups = (column) => {
       const layoutGroupView =
         !!column.galleryGroups.length &&
-        getVisibleItems(column.galleryGroups, container);
-      return (
-        layoutGroupView &&
-        layoutGroupView.map((group) => {
-          return group.rendered
-            ? React.createElement(GroupView, {
-                allowLoop:
-                  this.props.options.slideshowLoop &&
-                  this.props.galleryStructure.width >
-                    this.props.container.width,
-                ...group.renderProps(galleryConfig),
-                ariaHidden: group.idx > this.skipFromSlide,
-              })
-            : false;
-        })
-      );
-    };
+        this.getBufferedItems(column.galleryGroups, container);
+      if (layoutGroupView) {
+        return (
+          layoutGroupView.map(({group, shouldRender}) => {
+            return group.rendered
+              ? React.createElement(GroupView, {
+                  allowLoop:
+                    this.props.options.slideshowLoop &&
+                    this.props.galleryStructure.width >
+                      this.props.container.width,
+                  ...group.renderProps(galleryConfig),
+                  ariaHidden: group.idx > this.skipFromSlide,
+                  shouldRenderEmpty: !shouldRender,
```

Comment:
[question]
You could have simply returned only the `shouldRender`ed groups and then you could remove the ternary (`:` operator). What do you think?